app/services/teacher_training_public_api/sync_courses.rb

```diff
@@ -95,10 +103,14 @@ def add_accredited_provider(course, accredited_body_code, recruitment_cycle_year
     end
 
     def add_provider_relationship(course)
-      ProviderRelationshipPermissions.find_or_create_by!(
+      provider_relationship_permission = ProviderRelationshipPermissions.find_or_create_by!(
         training_provider: provider,
         ratifying_provider: course.accredited_provider,
       )
+
+      if !@incremental_sync && provider_relationship_permission.new_record?
```

Comment:
`.new_record?` won't return true as we would have already created the record at this point, if we find or initialise by, do the check then save we can raise the error appropriately.